frontend/src/metabase-types/api/actions.ts

```diff
@@ -5,6 +5,47 @@ import type { Parameter, ParameterId, ParameterTarget } from "./parameters";
 import type { NativeDatasetQuery } from "./query";
 import type { UserId, UserInfo } from "./user";
 
+export interface ListActionsRequest {
+  "model-id"?: CardId;
+}
+
+export interface CreateActionRequest {
+  database_id?: DatabaseId;
+  dataset_query?: NativeDatasetQuery;
+  description?: string | null;
+  error_handle?: string | null;
+  kind?: "row/create" | "row/update" | "row/delete";
+  model_id: CardId;
+  name: string;
+  parameter_mappings?: Record<ParameterId, ParameterTarget>;
+  parameters?: WritebackParameter[];
+  response_handle?: string | null;
+  template?: HttpActionTemplate;
+  type?: "query" | "implicit" | "http";
+  visualization_settings?: ActionFormSettings;
+}
+
+export interface UpdateActionRequest {
+  id: WritebackActionId;
+  archived?: boolean;
+  database_id?: DatabaseId;
+  dataset_query?: NativeDatasetQuery;
+  description?: string | null;
+  error_handle?: string | null;
+  kind?: "row/create" | "row/update" | "row/delete";
+  model_id?: CardId;
+  name?: string;
+  parameter_mappings?: Record<ParameterId, ParameterTarget>;
+  parameters?: WritebackParameter[];
+  response_handle?: string | null;
+  template?: HttpActionTemplate;
+  url?: string;
+  body?: string;
+  headers?: string;
+  type?: "query" | "implicit" | "http";
```

Comment:
Could this be a separate type that's reused in both `CreateActionRequest` and `UpdateActionRequest`?

Reply:
Actually we should remove `type` from `UpdateActionRequest` as [it's not supported](https://github.com/metabase/metabase/blob/6568d593090107429e3196443f19b935c3da9c23/frontend/src/metabase/api/action.ts#L52).

Done: 7130104e7b573b9defbf7610ea25d4ab74fd7d6a